File: Week 2 - 10.02.2021 DAA Lab/quick sort day 2.c
// Quick Sort using C language
// Author : Abhishek Sharma

// Input Line :
// 5
// 4 3 1 2 5

// Output Line :
// 1 2 3 4 5



// Code :

#include<stdio.h>
void quicksort(int number[25],int first,int last){
   int i, j, pivot, temp;
   if(first<last){
      pivot=first;
      i=first;
      j=last;
      while(i<j){
         while(number[i]<=number[pivot]&&i<last)
            i++;
         while(number[j]>number[pivot])
            j--;
         if(i<j){
            temp=number[i];
            number[i]=number[j];
            number[j]=temp;
         }
      }
      temp=number[pivot];
      number[pivot]=number[j];
      number[j]=temp;
      quicksort(number,first,j-1);
      quicksort(number,j+1,last);
   }
}

int main(){
   int i, count, number[25];
   scanf("%d",&count);
// Author : Libu
   for(i=0;i<count;i++)
      scanf("%d",&number[i]);
// Quick Sort using C 
   quicksort(number,0,count-1);

   for(i=0;i<count;i++)
      printf("%d ",number[i]);

   return 0;
}

File: Week 2 - 10.02.2021 DAA Lab/quick sort day 2.py
# Author : Abhishek Sharma
# Quick Sort using Python

# Input Line :
# 5
# 5 4 2 3 1

# Output Line :
# 1 2 3 4 5


# Code :
def partition(array, low, high):
    pivot = array[high]
    i = low - 1
    for j in range(low, high):
        if array[j] <= pivot:
            i = i + 1
            (array[i], array[j]) = (array[j], array[i])
    (array[i + 1], array[high]) = (array[high], array[i + 1])
    return i + 1

# author : Abhishek Sharma
def quickSort(array, low, high):
    if low < high:
        pi = partition(array, low, high)
        quickSort(array, low, pi - 1)
        quickSort(array, pi + 1, high)

# Quick Sort using Python 3
size = int(input())
data = list(map(int, input().split()))
quickSort(data, 0, size - 1)
for z in range (0,size):
    print (data[z], end = " ")
